addons/sale_timesheet/controllers/portal.py

```diff
@@ -63,6 +64,26 @@ def _get_groupby_mapping(self):
             invoice='timesheet_invoice_id')
         return groupby_mapping
 
+    def _task_get_page_view_values(self, task, access_token, **kwargs):
+        values = super()._task_get_page_view_values(task, access_token, **kwargs)
+        try:
+            if values['task'].sale_order_id and self._document_check_access('sale.order', values['task'].sale_order_id.id):
+                values['so_accessible'] = True
+            else:
+                values['so_accessible'] = False
+        except (AccessError, MissingError):
+            values['so_accessible'] = False
+
+        values['invoices_accessible'] = []
+        try:
+            if values['task'].sale_order_id and values['task'].sale_order_id.invoice_ids:
+                for invoice in values['task'].sale_order_id.invoice_ids:
+                    if(self._document_check_access('account.move', invoice.id)):
```

Comment:
Why are you testing the return value of `_document_check_access`? It returns the input document if that exists and is accessible, otherwise it raises an exception. There is no situation in which it returns a falsy value.